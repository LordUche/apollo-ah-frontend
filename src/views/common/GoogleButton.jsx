import React from 'react';
import PropTypes from 'prop-types';
import GoogleLogin from 'react-google-login';
import google from '../../images/icons/google.svg';
import ImageButton from './ImageButton';

/**
 * A function representing GoogleButton
 *
 * @param {{Function}} handleResponse - A response handler callback
 * @returns {JSX.Element} - DOM element
 */
const GoogleButton = ({ handleResponse }) => {
  return (
    <GoogleLogin
      clientId={process.env.GOOGLE_CLIENT_ID}
      autoLoad={false}
      render={renderProps => (
        <ImageButton
          imgSrc={google}
          onClick={renderProps.onClick}
          altText="login with google"
        />
      )}
      onSuccess={handleResponse}
      onFailure={handleResponse}
    />
  );
};

GoogleButton.propTypes = {
  handleResponse: PropTypes.func.isRequired,
};

export default GoogleButton;
