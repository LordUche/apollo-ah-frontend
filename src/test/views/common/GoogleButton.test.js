/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GoogleButton from '../../../views/common/GoogleButton';

describe('<GoogleButton />', () => {
  it('should render google button', () => {
    const enzymeWrapper = shallow(<GoogleButton handleResponse={jest.fn()} />);
    expect(enzymeWrapper.exists()).toBe(true);
    expect(toJson(enzymeWrapper)).toMatchSnapshot();
  });
});
