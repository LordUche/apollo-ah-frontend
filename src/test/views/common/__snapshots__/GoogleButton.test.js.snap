// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleButton /> should render google button 1`] = `
<e
  accessType="online"
  autoLoad={false}
  buttonText="Sign in with Google"
  cookiePolicy="single_host_origin"
  disabledStyle={
    Object {
      "opacity": 0.6,
    }
  }
  fetchBasicProfile={true}
  icon={true}
  isSignedIn={false}
  jsSrc="https://apis.google.com/js/api.js"
  onFailure={[MockFunction]}
  onRequest={[Function]}
  onSuccess={[MockFunction]}
  prompt=""
  render={[Function]}
  scope="profile email"
  tag="button"
  theme="light"
  type="button"
  uxMode="popup"
/>
`;
